.travis.yml

```diff
@@ -0,0 +1,50 @@
+language: python
+python:
+  - 3.9
+  - 3.10
+git:
+  depth: false
+
+# Ubuntu 20.04 (Focal) LTS
+os: linux
+dist: focal
+
+# Set the cache
+cache:
+  pip: true
+  directories:
+    - "$HOME/.cache/pre-commit"
+
+# Environment variables
+env:
+  global:
+    # Set Matplotlib backend to Agg (Non-interactive, only writes to files)
+    - MPLBACKEND=Agg
+
+# Installs poetry, pre-commit, tox, tox-travis
+before_install:
+  - curl -sSL https://install.python-poetry.org | python3 -
+  - poetry --version
+  - pip install coveralls
+  - pip install pre-commit
+
+# Installs all packages needed for tmi in a virtual environment
+install:
+  - poetry install
+  - pre-commit install-hooks
+
+# Run the linting (before_script) and run the tests
+before_script: pre-commit run --all-files
+script: poetry run pytest
+
+jobs:
+  include:
+    - stage: build
+      python: 3.10
```

Comment:
Per Noah's comment above, it's best that we don't build with Python 3.10 right now. I'm thinking 3.8 at latest.